src/core/image.js

```diff
@@ -506,6 +506,122 @@ var PDFImage = (function PDFImageClosure() {
       }
     },
 
+    /**
+     * Check whether resize image data or not
+     * @param {Number} comps Number of color components, 1 or 3 is supported.
+     * @param {Number} bpc Number of bits per component.
+     * @return {Boolean}
+     */
+    shallResizeImage: function PDFImage_shallResizeImage(comps, bpc) {
+        return (comps === 1 && (bpc === 1 || bpc === 8));
```

Comment:
Nit: Here, and elsewhere in the patch, please make sure that you indent using *two* spaces.